pytest_mozwebqa/cloud/saucelabs.py

```diff
@@ -7,124 +7,138 @@
-        requests.put('http://saucelabs.com//rest/v1/%s/jobs/%s' % (self.username, session),
-                     data=json.dumps(status), auth=(self.username, self.api_key))
-
-    def _video_html(self, session):
-        flash_vars = 'config={\
-            "clip":{\
-                "url":"https://assets.saucelabs.com/jobs/%(session)s/video.flv",\
-                "provider":"streamer",\
-                "autoPlay":false,\
-                "autoBuffering":true},\
-            "play": {\
-                "opacity":1,\
-                "label":null,\
-                "replayLabel":null},\
-            "plugins":{\
-                "streamer":{\
-                    "url":"https://cdn1.saucelabs.com/sauce_skin_deprecated/lib/flowplayer/flowplayer.pseudostreaming-3.2.13.swf",\
-                    "queryString":"%%3Fstart%%3D%%24%%7Bstart%%7D"},\
-                "controls":{\
-                    "mute":false,\
-                    "volume":false,\
-                    "backgroundColor":"rgba(0,0,0,0.7)"}},\
-            "playerId":"player%(session)s",\
-            "playlist":[{\
-                "url":"https://assets.saucelabs.com/jobs/%(session)s/video.flv",\
-                "provider":"streamer",\
-                "autoPlay":false,\
-                "autoBuffering":true}]}' % {'session': session}
-
-        return html.div(html.object(
-            html.param(value='true', name='allowfullscreen'),
-            html.param(value='always', name='allowscriptaccess'),
-            html.param(value='high', name='quality'),
-            html.param(value='#000000', name='bgcolor'),
-            html.param(
-                value=flash_vars.replace(' ', ''),
-                name='flashvars'),
-            width='100%',
-            height='100%',
-            type='application/x-shockwave-flash',
-            data='https://cdn1.saucelabs.com/sauce_skin_deprecated/lib/flowplayer/flowplayer-3.2.17.swf',
-            name='player_api',
-            id='player_api'),
-            id='player%s' % session,
-            style='border:1px solid #e6e6e6; float:right; height:240px;'
-                  'margin-left:5px; overflow:hidden; width:320px')
+name = 'Sauce Labs'
+
+
+def _get_config(option):
+    config = ConfigParser()
+    config.read('setup.cfg')
+    section = 'saucelabs'
+    if config.has_section(section) and config.has_option(section, option):
+        return config.get(section, option)
+
+
+def _credentials():
+    username = os.getenv('SAUCELABS_USERNAME', _get_config('username'))
+    api_key = os.getenv('SAUCELABS_API_KEY', _get_config('api-key'))
+    if username is None:
+        raise pytest.UsageError('Sauce Labs username must be set')
+    if api_key is None:
+        raise pytest.UsageError('Sauce Labs API key must be set')
+    return username, api_key
+
+
+def _tags():
+    tags = _get_config('tags')
+    if tags is not None:
+        return tags.split(',')
+    else:
+        return []
+
+
+def _split_class_and_test_names(nodeid):
```

Comment:
This method also seems to be duplicated in both this file and `browserstack.py`, which seems less than ideal. Splitting it off into a helper class seems like it might be a good idea.